Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -34,6 +34,26 @@ namespace Mantid {
 namespace NexusGeometry {
 namespace NexusGeometrySave {
 
+inline H5::Group safeCreateGroup(const H5::Group &parentGroup,
+                                 const std::string &childGroupName) {
+  H5std_string parentGroupName = H5_OBJ_NAME(parentGroup);
+  H5std_string intendedPath = parentGroupName + "/" + childGroupName;
+  for (hsize_t i = 0; i < parentGroup.getNumObjs(); ++i) {
+    if (parentGroup.getObjTypeByIdx(i) == GROUP_TYPE) {
+      H5std_string child = parentGroup.getObjnameByIdx(i);
+      H5std_string currentPath = parentGroupName + "/" + child;
```

Comment:
You shouldn't do this full path comparison. I don't understand why it is needed. I think your implementation is buggy anyway. Just compare the name directly.

Reply:
refactored